### PR TITLE
Fix ocp-index.1.1.4 for 4.03

### PR DESCRIPTION
**Pull request by lefessan**

### CODE REVIEW

**Comment by dsheets** [comment]
This fails on opam 1.1.2. You probably need something like `available: [opam-version >= "1.2"]`.

Also, the patch seems to be getting bigger... has it been upstreamed? Maybe it's time for a new upstream revision?

**Comment by lefessan** [comment]
@dsheets : in which files should I add such a constraint ? The only thing I did was to change some `ocaml-version: [ >= "4.02"]` into `available: [ ocaml-version >= "4.02"]`. Shall I revert these changes ?

I am currently upstreaming the patch, but for that, `ocp-index` needs to use a preprocessor because generic code cannot be compatible with all OCaml versions.

**Comment by dsheets** [comment]
Hmm, no I think it's fine and I was misreading the error message. Something seems to have broken with the checking for opam 1.1.2. Sorry for the confusion.

File: packages/ocp-index/ocp-index.0.1.0/opam [hunk 1]
```diff
@@ -17,5 +17,5 @@ depends: [
   "cmdliner"
 ]
 depopts: ["curses"]
-ocaml-version: [ >= "4.00.0" & < "4.02" ]
+available: [ocaml-version >= "4.00.0" & ocaml-version < "4.02" ]
```
Comment (samoht) [inline packages/ocp-index/ocp-index.0.1.0/opam]: @lefessan Can you add parentheses around the terms? opam 1.2.0 is a bit confused by that (I think), see https://travis-ci.org/ocaml/opam-repository/jobs/129653866#L880 This PR seems to be the first occurrence of that error in the logs.

Comment (samoht) [inline packages/ocp-index/ocp-index.0.1.0/opam]: e.g. 

```
available: [ (ocaml-version >= "4.00.0") & (ocaml-version < "4.02") ]
```

File: packages/ocp-index/ocp-index.1.0.0/opam [hunk 2]
```diff
@@ -22,7 +22,7 @@ depends: [
   "cmdliner"
 ]
 depopts: ["curses"]
-ocaml-version: [ >= "4.00.0" & < "4.02" ]
+available: [ ocaml-version >= "4.00.0" & available < "4.02" ]
```
Comment (samoht) [inline packages/ocp-index/ocp-index.1.0.0/opam]: actually the actual error is here: `available < "4.02" `

Comment (samoht) [inline packages/ocp-index/ocp-index.1.0.0/opam]: I'll make a PR

**Pull request merged**